### DISCUSSION TITLE
Support for Github Pages and static content generation

### DISCUSSION THREAD

**Discussion by paladini** [outline/outline#1986] [Ideas]
## Idea: Support for GH Pages / Static content generator

I was wondering if there's any official plans to allow static content generation from the current state of Outline localhost server in order to host at Github Pages, a free static content platform from Github that you all may be familiar with.

**Pros:**

- Users and open-source community: will make it possible to users to have a personal, self-hosted (but limited) experience with Outline for free at Github Pages. That can engage users and  
- Outline company: your solution will be more complete and let users use an unlimited and free application through an existing free service from Github. Besides that, it will also allow more people to get know your solution, consequently can increase your open-source community and your adoption in private companies that are managed or ran by developers from the open-source community.
- (please point out more pros in the comments)

**Cons:**

- Grow up your open-source community that contributes with issues, suggestions and PRs (??)
- (please point out more real cons in the comments)

## Reference / Background of the idea

I've developed a project called "[Ghost on Github Pages](https://github.com/paladini/ghost-on-github-pages)", where it's possible to generate a "picture"/snapshot of the running application in order to export the current state of the server to a rendered static content (HTML, CSS, JS, etc.). 

There's several other applications like that across the web, that generates static content from a localhost server application.  If you remember any project like this besides Ghost on Github Pages, please share in order to compare solutions.

If the maintainers of Outline project cannot implement something like that for some reason, but can point a way on how to do something like that with existing project, I could write some scripts to make that automatically in the future. Cannot promise anything, but have some interest in that.

**Comment by tommoor** [comment]
I think it's a good idea that I'm totally in support of and you could certainly build this project with the [API](https://www.getoutline.com/developers) quite easily 😄 

We get a lot of requests for the ability to share the entire knowledge base or a section of it publicly. This is definitely something that's on the roadmap but the exact implementation in terms of whether that would be static or not hasn't been worked out yet – it's unlikely that the feature we build internally would host elsewhere though as most requests we get expect it to work "out of the box".
